core/src/main/java/com/sequenceiq/cloudbreak/structuredevent/job/StructuredSynchronizerJob.java

```diff
@@ -0,0 +1,76 @@
+package com.sequenceiq.cloudbreak.structuredevent.job;
+
+import javax.inject.Inject;
+
+import org.quartz.DisallowConcurrentExecution;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.quartz.TracedQuartzJob;
+import com.sequenceiq.cloudbreak.service.stack.StackService;
+import com.sequenceiq.cloudbreak.structuredevent.LegacyDefaultStructuredEventClient;
+import com.sequenceiq.cloudbreak.structuredevent.StructuredSyncEventFactory;
+import com.sequenceiq.cloudbreak.structuredevent.event.StructuredSyncEvent;
+import com.sequenceiq.flow.core.FlowLogService;
+
+import io.opentracing.Tracer;
+
+@DisallowConcurrentExecution
+@Component
+public class StructuredSynchronizerJob extends TracedQuartzJob {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StructuredSynchronizerJob.class);
+
+    private String localId;
+
+    @Inject
+    private StackService stackService;
+
+    @Inject
+    private FlowLogService flowLogService;
+
+    @Inject
+    private StructuredSyncEventFactory structuredSyncEventFactory;
+
+    @Inject
+    private LegacyDefaultStructuredEventClient legacyDefaultStructuredEventClient;
+
+    public StructuredSynchronizerJob(Tracer tracer) {
+        super(tracer, "Structured Synchronizer Job");
+    }
+
+    @Override
+    protected Object getMdcContextObject() {
+        return stackService.getById(getStackId());
+    }
+
+    @Override
+    protected void executeTracedJob(JobExecutionContext context) throws JobExecutionException {
+        if (flowLogService.isOtherFlowRunning(getStackId())) {
```

Comment:
Why do we need this? Shouldn't we collect data if something is happening with the cluster?

Reply:
The goal is to periodically have info about clusters that are "not doing anything". Right now we can only have telemetry data in EDH about clusters that had something happened to them, like a stop/upgrade/start/etc event, but not about clusters that are "just running", so we cannot answer from this data that how many clusters are running/stopped/existing at a given moment.

Reply:
But shouldn't we collect clusters which stuck in a flow? Theoretically, it shouldn't happen, but it happened many times.